internal/core/openapi/openapi.go

```diff
@@ -81,33 +81,40 @@ func (openapi *Openapi) buildRouter() *gin.Engine {
 	router := gin.Default()
 	// build context for peer request
 	router.Use(openapi.buildContext)
+	// version 1 group
+	v1 := router.Group("/v1")
 	// login and logout
-	router.POST("/login", openapi.login)
-	router.GET("/logout", openapi.logout)
-	// router group
-	adminGroup := router.Group("/v1/user")
+	v1.POST("/login", openapi.login)
+	v1.GET("/logout", openapi.logout)
+	// user group
+	userGroup := v1.Group("/user")
 	{
 		// TODO test
-		adminGroup.POST("", openapi.user)
-		adminGroup.PUT("/:id", openapi.updateUser)
-		adminGroup.GET("/:id", openapi.userById)
-		adminGroup.DELETE("/:id", openapi.removeUser)
+		userGroup.POST("", openapi.user)
+		userGroup.PUT("/:id", openapi.updateUser)
+		userGroup.GET("/:id", openapi.userById)
+		userGroup.DELETE("/:id", openapi.removeUser)
+
+		// TODO envcd scopespace
+		userGroup.GET("/:userId/scopespace/:scopeSpaceId", openapi.scopeSpace)
+		userGroup.POST("/:userId/scopespace", openapi.putScopeSpace)
+		userGroup.PUT("/:userId/scopespace/:scopeSpaceId", openapi.updateScopeSpace)
+		userGroup.DELETE("/:userId/scopespace/:scopeSpaceId", openapi.removeScopeSpace)
+
+		// TODO envcd config
+		userGroup.GET("/:userId/scopespace/:scopeSpaceId/dict/:dictId", openapi.dictionary)
+		userGroup.POST("/:userId/scopespace/:scopeSpaceId/dict", openapi.putDictionary)
+		userGroup.PUT("/:userId/scopespace/:scopeSpaceId/dict/:dictId", openapi.updateDictionary)
+		userGroup.DELETE("/:userId/scopespace/:scopeSpaceId/dict/:dictId", openapi.removeDictionary)
 	}
-	envcdApplication := router.Group("/v1/app")
+	envcdScopeSpace := v1.Group("/scopespace")
```

Comment:
maybe `scopespaceGroup = v1.GET(/scopesapes)` and `dictionaryGroup = v1.GET(/dictionaries)` 

and zhe scopespace group link the user, ie, user's scopespace 

`
scopespaceGroup .GET("/user/:userId")
dictionaryGroup.GET("/user/:userId")
dictionaryGroup.GET("/user/:userId/scopespace/:scopespaceId")
`